tests/Cosmos/SignerTests.cpp

```diff
@@ -172,3 +173,135 @@ TEST(CosmosSigner, SignTxJson_WithMode) {
+    input.set_private_key(privateKey.data(), privateKey.size());
+
+    auto output = Signer::sign(input);
+
+    // real-world tx: https://www.mintscan.io/cosmos/txs/817101F3D96314AD028733248B28BAFAD535024D7D2C8875D3FE31DC159F096B
+    // curl -H 'Content-Type: application/json' --data-binary '{"tx_bytes": "Cr4BCr...1yKOU=", "mode": "BROADCAST_MODE_BLOCK"}' https://api.cosmos.network/cosmos/tx/v1beta1/txs 
+    // also similar TX: BCDAC36B605576C8182C2829C808B30A69CAD4959D5ED1E6FF9984ABF280D603
+    EXPECT_EQ(output.serialized(), "Cr4BCrsBCikvaWJjLmFwcGxpY2F0aW9ucy50cmFuc2Zlci52MS5Nc2dUcmFuc2ZlchKNAQoIdHJhbnNmZXISC2NoYW5uZWwtMTQxGg8KBXVhdG9tEgYxMDAwMDAiLWNvc21vczFta3k2OWNuOGVrdHd5MDg0NXZlYzl1cHNkcGhrdHh0MDNna3dseCorb3NtbzE4czBoZG5zbGxnY2Nsd2V1OWF5bXc0bmdrdHIyazBya3ZuN2ptbjIHCAEQgI6ZBBJoClAKRgofL2Nvc21vcy5jcnlwdG8uc2VjcDI1NmsxLlB1YktleRIjCiEC7O9c5DejAsZ/lUaN5LMfNukR9GfX5qUrQcHhPh1WNkkSBAoCCAEYAhIUCg4KBXVhdG9tEgUxMjUwMBCgwh4aQK0HIWdFMk+C6Gi1KG/vELe1ffcc1aEWUIqz2t/ZhwqNNHxUUSp27wteiugHEMVTEIOBhs84t2gIcT/nD/1yKOU=");
+    EXPECT_EQ(hex(output.signature()), "ad07216745324f82e868b5286fef10b7b57df71cd5a116508ab3dadfd9870a8d347c54512a76ef0b5e8ae80710c55310838186cf38b76808713fe70ffd7228e5");
+    EXPECT_EQ(output.json(), "");
+    EXPECT_EQ(output.error(), "");
+}
+
+TEST(CosmosSigner, SignIbcTransferJson) {
+    auto input = Proto::SigningInput();
+    input.set_signing_mode(Proto::JSON);
+    input.set_account_number(546179);
+    input.set_chain_id("cosmoshub-4");
+    input.set_memo("");
+    input.set_sequence(2);
+
+    Address fromAddress;
+    EXPECT_TRUE(Address::decode("cosmos1mky69cn8ektwy0845vec9upsdphktxt03gkwlx", fromAddress));
+    Address toAddress;
+    EXPECT_TRUE(Address::decode("osmo18s0hdnsllgcclweu9aymw4ngktr2k0rkvn7jmn", toAddress));
+
+    auto msg = input.add_messages();
+    auto& message = *msg->mutable_transfer_tokens_message();
+    message.set_source_port("transfer");
+    message.set_source_channel("channel-141");
+    message.set_sender(fromAddress.string());
+    message.set_receiver(toAddress.string());
+    message.mutable_token()->set_denom("uatom");
+    message.mutable_token()->set_amount(100000); // 0.1 ATOM
+    message.set_timeout_height_revision_number(1);
+    message.set_timeout_height_revision_height(8800000);
+    message.set_timeout_timestamp(0);
+
+    auto& fee = *input.mutable_fee();
+    fee.set_gas(500000);
+    auto amountOfFee = fee.add_amounts();
+    amountOfFee->set_denom("uatom");
+    amountOfFee->set_amount(12500);
+
+    auto privateKey = parse_hex("8bbec3772ddb4df68f3186440380c301af116d1422001c1877d6f5e4dba8c8af");
+    EXPECT_EQ(Cosmos::Address(TWCoinTypeCosmos, PrivateKey(privateKey).getPublicKey(TWPublicKeyTypeSECP256k1)).string(), "cosmos1mky69cn8ektwy0845vec9upsdphktxt03gkwlx");
+    input.set_private_key(privateKey.data(), privateKey.size());
+
+    auto output = Signer::sign(input);
+
+    assertJSONEqual(output.json(), R"(
```

Comment:
Does endpoint support json format?

Reply:
I think not, I added json test for coverage.  But maybe I should remove json support completely.

Reply:
yes, let's just keep json as legacy code, no new features